<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>967</width>
    <height>660</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_4">
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QWidget" name="gridLayoutWidget">
       <layout class="QGridLayout" name="electrodeLayout" rowstretch="0" columnstretch="0">
        <property name="sizeConstraint">
         <enum>QLayout::SetDefaultConstraint</enum>
        </property>
        <property name="leftMargin">
         <number>10</number>
        </property>
        <property name="topMargin">
         <number>10</number>
        </property>
        <property name="rightMargin">
         <number>10</number>
        </property>
        <property name="bottomMargin">
         <number>10</number>
        </property>
        <property name="spacing">
         <number>50</number>
        </property>
       </layout>
      </widget>
      <widget class="QWidget" name="layoutWidget">
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <property name="sizeConstraint">
         <enum>QLayout::SetFixedSize</enum>
        </property>
        <item>
         <widget class="QTabWidget" name="ModeButtonTab">
          <property name="currentIndex">
           <number>2</number>
          </property>
          <widget class="QWidget" name="DropletMode">
           <attribute name="title">
            <string>Droplet Mode</string>
           </attribute>
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <item>
               <widget class="QPushButton" name="DispenceButton">
                <property name="statusTip">
                 <string>Perform Dispence</string>
                </property>
                <property name="text">
                 <string>Dispense</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_3">
              <item>
               <widget class="QPushButton" name="SplitButton">
                <property name="statusTip">
                 <string>Perform Split</string>
                </property>
                <property name="text">
                 <string>Split</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_4">
              <item>
               <widget class="QPushButton" name="PreviewButton">
                <property name="statusTip">
                 <string>Preview current sequence</string>
                </property>
                <property name="text">
                 <string>Preview</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QPushButton" name="CancelPreviwButton">
              <property name="statusTip">
               <string>Cancel the preview sequence</string>
              </property>
              <property name="styleSheet">
               <string notr="true">background: rgb(255, 79, 48);</string>
              </property>
              <property name="text">
               <string>Cancel Preview</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout">
              <item>
               <widget class="QPushButton" name="StartButton">
                <property name="statusTip">
                 <string>Send the path sequence to Arduino</string>
                </property>
                <property name="styleSheet">
                 <string notr="true">background: rgb(71, 200, 51)</string>
                </property>
                <property name="text">
                 <string>Start</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QPushButton" name="CancelStartButton">
              <property name="statusTip">
               <string>Stop sequencing</string>
              </property>
              <property name="styleSheet">
               <string notr="true">background: rgb(255, 79, 48);</string>
              </property>
              <property name="text">
               <string>Cancel</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="addDrop">
              <property name="statusTip">
               <string>Add a droplet to an electrode</string>
              </property>
              <property name="text">
               <string>Add Droplet</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="removedrop">
              <property name="statusTip">
               <string>Remove a droplet from an electrode</string>
              </property>
              <property name="text">
               <string>Remove Droplet</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="BeginButton">
              <property name="statusTip">
               <string>Press when done selecting electrodes</string>
              </property>
              <property name="text">
               <string>Begin Dispensing</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="CancelButton">
              <property name="toolTip">
               <string>[Esc]</string>
              </property>
              <property name="toolTipDuration">
               <number>-1</number>
              </property>
              <property name="statusTip">
               <string>Press to cancel the action</string>
              </property>
              <property name="styleSheet">
               <string notr="true">background: rgb(255, 79, 48);</string>
              </property>
              <property name="text">
               <string>Cancel</string>
              </property>
              <property name="shortcut">
               <string>Esc</string>
              </property>
              <property name="autoDefault">
               <bool>false</bool>
              </property>
              <property name="default">
               <bool>false</bool>
              </property>
              <property name="flat">
               <bool>false</bool>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="ElectrodeMode">
           <attribute name="title">
            <string>Electrode Mode</string>
           </attribute>
           <layout class="QVBoxLayout" name="verticalLayout_11">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_10">
              <item>
               <widget class="QPushButton" name="Increment_EmodeButton">
                <property name="toolTip">
                 <string/>
                </property>
                <property name="statusTip">
                 <string>Press spacebar to Increment</string>
                </property>
                <property name="text">
                 <string>Increment</string>
                </property>
                <property name="shortcut">
                 <string>Space</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="preview_EmodeButton">
                <property name="statusTip">
                 <string>Preview the sequence</string>
                </property>
                <property name="text">
                 <string>Preview</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="CancelPreviwEmodeButton">
                <property name="statusTip">
                 <string>Cancel the preview sequence</string>
                </property>
                <property name="styleSheet">
                 <string notr="true">background: rgb(255, 79, 48);</string>
                </property>
                <property name="text">
                 <string>Cancel Preview</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="Start_EmodeButton">
                <property name="statusTip">
                 <string>Send the path sequence to Arduino</string>
                </property>
                <property name="styleSheet">
                 <string notr="true">background: rgb(71, 200, 51)</string>
                </property>
                <property name="text">
                 <string>Start</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="CancelStart_EmodeButton">
                <property name="statusTip">
                 <string>Stop the sequence</string>
                </property>
                <property name="styleSheet">
                 <string notr="true">background: rgb(255, 79, 48);</string>
                </property>
                <property name="text">
                 <string>Cancel</string>
                </property>
               </widget>
              </item>
              <item>
               <layout class="QVBoxLayout" name="verticalLayout_7">
                <item>
                 <layout class="QHBoxLayout" name="horizontalLayout_6">
                  <item>
                   <widget class="QCheckBox" name="turnOn_EmodeButton">
                    <property name="statusTip">
                     <string>Turn on an electrode</string>
                    </property>
                    <property name="text">
                     <string>Turn ON Electrode</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QCheckBox" name="turnOff_EmodeButton">
                    <property name="statusTip">
                     <string>Turn off an electrode</string>
                    </property>
                    <property name="text">
                     <string>Turn OFF Electrode</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item>
                 <widget class="QCheckBox" name="RealTimeActuationBox">
                  <property name="statusTip">
                   <string>Actuate Electrode in Real-Time</string>
                  </property>
                  <property name="text">
                   <string>Real-Time Actuation</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="tab">
           <attribute name="title">
            <string>Parameters</string>
           </attribute>
           <layout class="QVBoxLayout" name="verticalLayout_12">
            <item>
             <widget class="QLabel" name="label_8">
              <property name="statusTip">
               <string>Set the number of iterations for a sequence</string>
              </property>
              <property name="text">
               <string>Sequence Iterations</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="IterationspinBox">
              <property name="minimum">
               <number>1</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_9">
              <property name="statusTip">
               <string>Set the delay between iterations</string>
              </property>
              <property name="text">
               <string>Iteration Delay (ms)</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="IterationDelayText">
              <property name="text">
               <string>2000</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_10">
              <property name="statusTip">
               <string>Set the amount of time the electrode is turned on</string>
              </property>
              <property name="text">
               <string>Actuation Time (ms)</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_8">
              <item>
               <widget class="QLineEdit" name="ActuationTimeText">
                <property name="text">
                 <string>1000</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="setActuationButton">
                <property name="statusTip">
                 <string>Send the actuation delay instruction to Arduino</string>
                </property>
                <property name="text">
                 <string>Set</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>Instruction Monitor</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QTextEdit" name="InstructonMonitor"/>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QTabWidget" name="ModeTableTab">
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="DropletMode">
       <attribute name="title">
        <string>Droplet Mode</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_6">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_5">
            <item>
             <widget class="QLabel" name="label">
              <property name="text">
               <string>Maximum Step:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="dropTime">
              <property name="enabled">
               <bool>true</bool>
              </property>
              <property name="buttonSymbols">
               <enum>QAbstractSpinBox::UpDownArrows</enum>
              </property>
              <property name="maximum">
               <number>500</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_2">
              <property name="text">
               <string>Current Step:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="currentStepText">
              <property name="text">
               <string>TextLabel</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_2">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QTableWidget" name="dropTable">
            <property name="rowCount">
             <number>3</number>
            </property>
            <property name="columnCount">
             <number>3</number>
            </property>
            <attribute name="horizontalHeaderVisible">
             <bool>false</bool>
            </attribute>
            <attribute name="verticalHeaderVisible">
             <bool>false</bool>
            </attribute>
            <row/>
            <row/>
            <row/>
            <column/>
            <column/>
            <column/>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="ElectrodeMode">
       <attribute name="title">
        <string>Electrode Mode</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_9">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_8">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_7">
            <item>
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>Maximum Step:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="dropTimeEmode">
              <property name="enabled">
               <bool>true</bool>
              </property>
              <property name="buttonSymbols">
               <enum>QAbstractSpinBox::UpDownArrows</enum>
              </property>
              <property name="maximum">
               <number>500</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_6">
              <property name="text">
               <string>Current Step:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="currentStepText_Emode">
              <property name="text">
               <string>TextLabel</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_3">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QTableWidget" name="dropTableEmode">
            <property name="rowCount">
             <number>3</number>
            </property>
            <property name="columnCount">
             <number>2</number>
            </property>
            <attribute name="horizontalHeaderVisible">
             <bool>false</bool>
            </attribute>
            <attribute name="verticalHeaderVisible">
             <bool>false</bool>
            </attribute>
            <row/>
            <row/>
            <row/>
            <column/>
            <column/>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>967</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="New_Layout"/>
    <addaction name="Save_Layout"/>
    <addaction name="Open_Layout"/>
   </widget>
   <widget class="QMenu" name="menuSequencing">
    <property name="title">
     <string>Sequencing</string>
    </property>
    <addaction name="Save_Sequence"/>
    <addaction name="Open_Sequence"/>
   </widget>
   <widget class="QMenu" name="menuAdruino">
    <property name="title">
     <string>Adruino</string>
    </property>
    <addaction name="Connect"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuSequencing"/>
   <addaction name="menuAdruino"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="New_Layout">
   <property name="text">
    <string>New Layout</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="Save_Layout">
   <property name="text">
    <string>Save Layout</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="Open_Layout">
   <property name="text">
    <string>Open Layout</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="Save_Sequence">
   <property name="text">
    <string>Save Sequence</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="Open_Sequence">
   <property name="text">
    <string>Open Sequence</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+O</string>
   </property>
  </action>
  <action name="Connect">
   <property name="text">
    <string>Connect </string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+C</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
